src/toil/test/src/jobCacheTest.py

```diff
@@ -1125,6 +1126,106 @@ def testBatchSystemCleanupCanHandleWorkerDeaths(self):
             assert not os.path.exists(nonLocalFile1)
             assert not os.path.exists(nonLocalFile2)
 
+        # A test to check if files created written to the jobstore on a failed job are deleted or
+        # not.
+        def testOrphanedFilesAreDeletedFromJobStore(self):
+            """
+            Run a job that creates a file, writes it to the file store and records the jobstore file
+            ID to an external file before dying.  Ensure that the file is deleted from the jobstore.
+            :return:
+            """
+            # There can be no retries
+            self.options.retryCount = 0
+            self.options.logLevel = 'DEBUG'
+            workdir = self._createTempDir(purpose='nonLocalDir')
+            nonLocalFile = os.path.join(workdir, str(uuid4()))
+            # The first file has to exist and be empty
+            file1 = open(nonLocalFile, 'w').close()
+            assert os.path.exists(nonLocalFile)
+            root = Job()
+            A = Job.wrapJobFn(self._testOrphanedFilesAreDeletedFromJobStore_A,
+                              nonLocalFile=nonLocalFile)
+            B = Job.wrapJobFn(self._testOrphanedFilesAreDeletedFromJobStore_B,
+                              nonLocalFile=nonLocalFile)
+            C = Job.wrapJobFn(self._testOrphanedFilesAreDeletedFromJobStore_C,
+                              nonLocalFile=nonLocalFile)
+            root.addChild(A)
+            root.addChild(B)
+            B.addChild(C)
+            try:
+                Job.Runner.startToil(root, self.options)
+            except FailedJobsException as err:
+                self.assertEqual(err.numberOfFailedJobs, 2) # root and A
+            assert not os.path.exists(nonLocalFile)
+
+        @staticmethod
```

Comment:
We can use static methods as job functions? I seem to remember that we used to have to alias those as global variable at the module level. Or did we automate the aliasing?